src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -440,7 +481,11 @@ public void DoWriteIfNeeded()
                 }, this);
 
                 Self._head = _lockedEnd.Block;
-                Self._head.Start = _lockedEnd.Index;
```

Comment:
Fixed it but there would be a recursion error as `WriteAllPending` is called at the end of `OnWriteComplete` so when the stack unwinds _head could be set to an old value. Mitigating...